elastic/beats
# Add agent fleet enrolment k8s manifest

## What does this PR do?

This PR provides the proposed manifests  to start agent as Daemonset and automatically enrol the pods to fleet server.


## Why is it important?

Users need a proposed way to deploy agent pods as daemonset in their cluster and automatically enrol them to the feet server. Manually enroling agents through Kibana UI is not needed anymore.

## Checklist




## How to test this PR locally

Spin up an elastic stack with fleet server. Then configure properly **elastic-agent-managed-kubernetes.yaml** with the 
right values for **FLEET_URL**, **KIBANA_HOST**, **KIBANA_FLEET_USERNAME**,  **KIBANA_FLEET_PASSWORD** and apply the manifest in a kubernetes cluster.
Notice in Kibana UI-->Management-->Fleet-->Agents section the new agents automatically enroling to fleet.
## Related issues


- Closes #25237
